src/tls.c

```diff
@@ -9261,6 +9261,11 @@ static int TLSX_KeyShare_SetSupported(WOLFSSL* ssl)
     for (; curve != NULL; curve = curve->next) {
         if (!TLSX_KeyShare_IsSupported(curve->name))
             continue;
+    #ifdef OPENSSL_EXTRA
+        /* Check if server supports group. */
```

Comment:
In `TLSX_ValidateSupportedCurves` we also check curve->name < `WOLFSSL_ECC_X448`. Should we also add it here?

```
/* skip if name is not in supported ECC range */
if (curve->name > WOLFSSL_ECC_X448)
    continue;
```

Reply:
@icing our supplementary testing turned up an annoying little nit in #5471 when building with `g++` as the compiler.  yes we have customer(s) that do that for some reason...

can you fix this in this PR?

```
src/tls.c: In function ‘int TLSX_QuicTP_Use(WOLFSSL*, TLSX_Type, int)’:
3434210b13 (<stefan.eissing@greenbytes.de> 2022-08-16 18:13:25 +0200 10449)         QuicTransportParam_free(extension->data, ssl->heap);
src/tls.c:10449:44: error: invalid conversion from ‘void*’ to ‘const QuicTransportParam*’ [-fpermissive]
10449 |         QuicTransportParam_free(extension->data, ssl->heap);
      |                                 ~~~~~~~~~~~^~~~
      |                                            |
      |                                            void*
In file included from src/tls.c:33:
4431438fb2 (<stefan.eissing@greenbytes.de> 2022-08-08 13:24:00 +0200 3547) WOLFSSL_LOCAL void QuicTransportParam_free(const QuicTransportParam *tp, void *heap);
./wolfssl/internal.h:3547:70: note:   initializing argument 1 of ‘void QuicTransportParam_free(const QuicTransportParam*, void*)’
 3547 | WOLFSSL_LOCAL void QuicTransportParam_free(const QuicTransportParam *tp, void *heap);
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~^~
```

Reply:
> In `TLSX_ValidateSupportedCurves` we also check curve->name < `WOLFSSL_ECC_X448`. Should we also add it here?
> 
> ```
> /* skip if name is not in supported ECC range */
> if (curve->name > WOLFSSL_ECC_X448)
>     continue;
> ```

Good question. In the recent push, I made the disabled checking safe in a common function that checks the curve to be in range. So we have that everywhere.

The special case for curves `> WOLFSSL_ECC_MAX` in `TLSX_ValidateSupportedCurves()` excludes PQC and the FFDHE ones. The check in `TLSX_KeyShare_Establish()` excludes *only* the PQC ones.

I am not sure which condition we'd want in `TLSX_KeyShare_SetSupported()`. I'd be happy to add the `> WOLFSSL_ECC_MAX` if that is the correct thing. @SparkiDev ?